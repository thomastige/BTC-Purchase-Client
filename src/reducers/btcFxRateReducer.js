import * as actionType from '../actions/ActionType';

const btcFxRateReducer = (state = 0, action) => {
  let newState;
  switch (action.type) {
    case actionType.SET_BTC_FXRATE:
		
      return newState = state + action.payload;
    default:
      return state
  }
}

export default btcFxRateReducer;