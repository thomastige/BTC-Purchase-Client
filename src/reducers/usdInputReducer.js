import * as actionType from '../actions/ActionType';

const usdInputReducer = (state = 0, action) => {
  let newState;
  switch (action.type) {
    case actionType.SET_USD_INPUT:
      return newState = action.payload;
    default:
      return state
  }
}

export default usdInputReducer;