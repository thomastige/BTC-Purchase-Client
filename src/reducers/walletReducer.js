import * as actionType from '../actions/ActionType';

const walletReducer = (state = 0, action) => {
	let newState;
	switch (action.type) {
		case actionType.INITIALIZE_WALLET:
		return newState = action.payload; 
	case actionType.UPDATE_WALLET:
		return newState = action.payload;
	default:
		return state
	}
}

export default walletReducer;