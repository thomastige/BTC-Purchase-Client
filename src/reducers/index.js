import { combineReducers } from 'redux';
import btcFxRateReducer from './btcFxRateReducer';
import usdInputReducer from './usdInputReducer';
import walletReducer from './walletReducer';

const reducers = combineReducers({
  btcFxRateReducer,
  usdInputReducer,
  walletReducer
})

export default reducers