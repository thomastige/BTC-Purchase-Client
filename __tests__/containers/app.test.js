import React from 'react';
import ReactDOM from 'react-dom';
import App from 'src/App';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducers from 'reducers';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store'


Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureStore();
let store;


const fetchPromise = Promise.resolve({
	json: () => Promise.resolve({last_price: 0.12345})
});
global.fetch = () => fetchPromise;

describe('App', () => {
    it('renders without crashing', () => {
		const store = createStore(reducers);
        const div = document.createElement('div');
        ReactDOM.render(
			<Provider store={store}>
				<App/>
			</Provider>
		, div);
    });
});

test('Testing Snapshots', () => {
	const shallowWrapper = shallow(<App/>);
    expect(shallowWrapper).toMatchSnapshot();
	store = mockStore({walletReducer: {usd: 200, btc:0}});
	const mountWrapper = mount(<Provider store={store}><App/></Provider>);
    expect(mountWrapper).toMatchSnapshot();
});

test('Validate link between USD and BTC inputs', () => {
	store = mockStore({usdInputReducer: 1, btcFxRateReducer: 0.12345, walletReducer: {usd: 200, btc:0}});
	const wrapper = mount(<Provider store={store}><App/></Provider>);
	expect(wrapper.find('input').at(3).props().value).toBe(0.12345)
});