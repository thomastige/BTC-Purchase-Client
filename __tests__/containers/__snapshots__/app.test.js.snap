// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Snapshots 1`] = `
<div
  id="root"
>
  <Connect(BtcFxRate) />
  <Connect(Wallet) />
  <Connect(InputForm) />
</div>
`;

exports[`Testing Snapshots 2`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <App>
    <div
      id="root"
    >
      <Connect(BtcFxRate)>
        <BtcFxRate
          actions={[Function]}
          dispatch={[Function]}
        >
          <div
            hidden="true"
          />
        </BtcFxRate>
      </Connect(BtcFxRate)>
      <Connect(Wallet)>
        <Wallet
          btcValue="0.00000000"
          dispatch={[Function]}
          usdValue="200.00"
        >
          <div>
            <p>
              Account Balance
            </p>
            <table
              id="wallet"
            >
              <tbody>
                <tr>
                  <td
                    className="walletRowTitle"
                  >
                    USD
                  </td>
                  <td>
                    200.00
                  </td>
                </tr>
                <tr>
                  <td
                    className="walletRowTitle"
                  >
                    BTC
                  </td>
                  <td>
                    0.00000000
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </Wallet>
      </Connect(Wallet)>
      <Connect(InputForm)>
        <InputForm
          actions={[Function]}
          btcValue={0}
          dispatch={[Function]}
          usdValue={200}
        >
          <div>
            <p>
              Trade
            </p>
            <table>
              <tbody>
                <tr>
                  <td>
                    <Connect(UsdInput)>
                      <UsdInput
                        actions={[Function]}
                        dispatch={[Function]}
                        placeholderText="Fetching Exchange Rates..."
                      >
                        <div>
                          <input
                            className="inputLabel"
                            id="UsdInputLabel"
                            readOnly={true}
                            value="USD"
                          />
                          <br />
                          <br />
                          <input
                            id="usdInput"
                            onChange={[Function]}
                            placeholder="Fetching Exchange Rates..."
                          />
                        </div>
                      </UsdInput>
                    </Connect(UsdInput)>
                  </td>
                </tr>
              </tbody>
            </table>
            <p>
              For
            </p>
            <table>
              <tbody>
                <tr>
                  <td>
                    <Connect(BtcInput)>
                      <BtcInput
                        btcValue=""
                        dispatch={[Function]}
                      >
                        <div>
                          <input
                            className="inputLabel"
                            readOnly={true}
                            value="BTC"
                          />
                          <br />
                          <br />
                          <input
                            id="btcDisplay"
                            placeholder="Display Quote"
                            readOnly={true}
                            value=""
                          />
                        </div>
                      </BtcInput>
                    </Connect(BtcInput)>
                  </td>
                </tr>
                <tr>
                  <td>
                    <button
                      id="tradeButton"
                      onClick={[Function]}
                    >
                      Trade
                    </button>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </InputForm>
      </Connect(InputForm)>
    </div>
  </App>
</Provider>
`;
